File: model/src/main/java/com/blackopalsolutions/goneviral/model/request/UpdateUserRequest.java
package com.blackopalsolutions.goneviral.model.request;

import com.blackopalsolutions.goneviral.model.domain.User;

public class UpdateUserRequest {
    private User user;

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }
}

File: shared/src/commonMain/kotlin/com/blackopalsolutions/goneviral/model/request/UpdateUserRequest.kt
package com.blackopalsolutions.goneviral.model.request

import com.blackopalsolutions.goneviral.model.domain.User

class UpdateUserRequest(val user: User) : Request() {

    override fun encodeToJsonString(): String =
        "{user:{" +
                "\"id\":${user.id}," +
                "\"username\":${user.username}," +
                "\"password\":${user.password}," +
                "\"gamesWon\":${user.gamesWon}," +
                "\"gamesLost\":${user.gamesLost}}}"
}